packages/core/src/model.js

```diff
@@ -857,7 +857,7 @@ ${associationOwner._getAssociationDebugList()}`);
     const missingIndexes = this.getIndexes()
       .filter(item1 => !existingIndexes.some(item2 => item1.name === item2.name))
       .sort((index1, index2) => {
-        if (this.sequelize.options.dialect === 'postgres') {
+        if (this.sequelize.options.dialect === 'postgres' || this.sequelize.options.dialect === 'cockroachdb') {
```

Comment:
Following merge of main in this PR, keep in mind that `this.sequelize.options.dialect` should not be used anymore. Use `this.sequelize.dialect.name` instead